csq.py

```diff
@@ -1,41 +1,256 @@
+        exit(1)
+
+    if version_info.minor < 10:
+        print("Error: Python version 3.10 or above required")
+        exit(1)
+
+def printHelp(exit_code = 1):
+    """Print help
+        Print the help message for the program
+    """
+
+    print("Usage: csq [options] <file>")
+    print("Options:")
+    print("  This options are to be used when compiling the cpp code generated by csq")
+    print("  -h, --help\t\t\tPrint this help message")
+    print("  -v, --version\t\t\tPrint the version of csq")
+    print("  -u, --uninstall\t\tUninstall the csq")
+    print("  -k, --keep\t\t\tKeep the generated cpp file")
+    print("  -o, --output <file>\t\tSpecify the output file")
+    print("  -O, --optimize <level>\tSpecify the optimization level")
+    print("  -c, --compile\t\t\tCompile the file and generate object file")
+    print("  -S, --assembly\t\tGenerate assembly code")
+    print("  -E, --preprocess\t\tGenerate preprocessed code")
+    print("  -g, --debug\t\t\tGenerate debug symbols")
+
+    exit(exit_code)
+
+def isFileValid(file):
+    """Check if the file is valid
+        Check if the arguments passed are enough or and valid
+    """
+
+    if len(arguments) < 2:
+        printHelp()
+        return False
+
+    # Check if the file exists
+    try:
+        open(file, "r")
+    except FileNotFoundError:
+        printHelp()
+        return False
+
+    return True
+
+def findIncludePath():
+    """Find the include path
+        The function finds the include path for the csq compiler
+        Check for predefined locations instead of just relying on environment variables
+        Sometimes the environment variables are not set (or shell incorrectly configured)
+
+    Check for the following locations:
+        1. $HOME/.csq/include/csq
+        2. /usr/local/include/csq
+        3. /opt/csq
+        4. /usr/include/csq
+        4. Current directory (Maybe a developer is testing the compiler)
+    """
+
+    # Check for $HOME/.csq/include/csq
+    home = getenv("HOME")
+    if home is not None:
+        csq_include_path = home + "/.csq/include/csq"
+        if path.exists(csq_include_path):
+            return csq_include_path
+
+    # Check for /usr/local/include/csq
+    if path.exists("/usr/local/include/csq"):
+        return "/usr/local/include/csq"
+
+    # Check for /opt/csq
+    if path.exists("/opt/csq"):
+        return "/opt/csq"
+
+    # Check for /usr/include/csq
+    if path.exists("/usr/include/csq"):
+        return "/usr/include/csq"
+
+    # Check for current directory
+    if path.exists(pwd()):
+        return pwd()
+
+    # If none of the above locations are found, return None
+    return None
+
+def compileFile(file, options, keep = False):
+    """Compile the code in a file
+        The function takes in a file as string and compiles it.
+        The compiled code is stored in a file with the same name
+    """
+    # csq include path path
+    csq_include_path = findIncludePath()
```

Comment:
Instead of it use 
```
csq_include_path = getenv("CSQ_INCLUDE")
```
Because we need to use global csq path in the system instead of current path.